v4/export/ir_impl.go

```diff
@@ -51,6 +51,117 @@ func (iter *rowIter) HasNext() bool {
 	return iter.hasNext
 }
 
+// rowIter implements the SQLRowIter interface.
+// Note: To create a rowIter, please use `newRowIter()` instead of struct literal.
+type rowIterChunks struct {
+	tctx    *tcontext.Context
+	conn    *sql.Conn
+	rows    *sql.Rows
+	hasNext bool
+	id      int
+	queries []string
+	args    []interface{}
+	err     error
+}
+
+func newRowIterChunks(tctx *tcontext.Context, conn *sql.Conn, queries []string, argLen int) *rowIterChunks {
+	r := &rowIterChunks{
+		tctx:    tctx,
+		conn:    conn,
+		queries: queries,
+		id:      0,
+		args:    make([]interface{}, argLen),
+	}
+	r.nextRows()
```

Comment:
I was wandering why this iter needs an extra next. will review later

Reply:
Because some sqls won't return result.